obspy/clients/httpproxy.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+"""
+Establish a socket connection through an HTTP proxy.
+Author: Fredrik Østrem <frx.apps@gmail.com>
+License:
+  This code can be used, modified and distributed freely, as long as it is this
+  note containing the original author, the source and this license, is put
+  along with the source code.
+
+J. MacCarthy, modified from https://gist.github.com/frxstrem/4487802
+
+"""
+from __future__ import (absolute_import, division, print_function,
+                        unicode_literals)
+from future.builtins import *  # NOQA
+
+import socket
+from base64 import b64encode
+import urllib
```

Comment:
You will have to import urllib with future to make sure it works across Python 2 and 3, see e.g. here: https://github.com/obspy/obspy/blob/ed0c7fb0074f657e807577e88466680c53f195ff/obspy/clients/fdsn/client.py#L29-L34

I think this is not a nice way and I hope we'll get rid of `future` eventually but for now it is needed.